include/deal.II/multigrid/mg_transfer.templates.h

```diff
@@ -399,6 +391,17 @@ MGLevelGlobalTransfer<VectorType>::set_component_to_block_map(
   component_to_block_map = map;
 }
 
+template <typename VectorType>
+template <int dim, int spacedim>
+void
+MGLevelGlobalTransfer<VectorType>::assert_built(
+  const DoFHandler<dim, spacedim> &dof_handler) const
```

Comment:
And this also looks correct to me.

Reply:
Actually, I think that this is where you want to have the `inline DEAL_II_ALWAYS_INLINE`.

Reply:
I don't think that we have to force the compiler to inline this and would just remove `DEAL_II_ALWAYS_INLINE`.

Reply:
If the function should be declared `inline` at all is a bit a question of philosophy. Templated functions don't require to be declared `inline` at all. On the other hand, if `*templates.h` are thought to be used the same way as `*.h` files and logically it makes sense to have them `inline` to be consistent with non-templated classes.

In this file, no other function is declared `inline` so I would rather omit it for consistency but I don't feel strongly about this.

Reply:
That's all fair enough :-) I don't feel adamant about it one way or the other -- I suspect that the compiler will just do the right thing, and if it doesn't, then the call to the function is still simply a jump to a `ret` instruction ;-)

Reply:
Of course. It's just a question of style. :slightly_smiling_face: